src/modules/site-v2/templates/data/submit-trait-form.html

```diff
@@ -0,0 +1,185 @@
+    </form>
+</div> <!-- /Container -->
+
+{% endblock %}
+
+{% block script %}
+
+<script>
+    {% include '_scripts/utils.js' %} {#/* defines: flash_message */#}
+
+    $(document).ready(function() {
+        
+        // Add 'required' class to labels
+        $('.label-group').each(function() {
+            const requiredSelector = $(this).next().attr('required') && $(this).find('label')
+
+            if (requiredSelector) {
+                requiredSelector.addClass('required')
+            }
+            if ($(this).children('label').text().includes('file')) {
+                $(this).children('label').addClass('required')
+            }
+        })
+
+        $.ajaxSetup({
+            beforeSend: function(xhr, settings) {
+                if (!/^(GET|HEAD|OPTIONS|TRACE)$/i.test(settings.type) && !this.crossDomain) {
+                xhr.setRequestHeader("X-CSRFToken", "{{ form.csrf_token._value() }}")
+                }
+            }
+        });
+
+        $('#speciesSelect, #file').on('change', function() {
+            $('#file-upload').prop('disabled', !$('#speciesSelect').val() || !$('#file').val())
+        })
+
+        $('#file').on('change', function() {
+            $('#file-content').empty()
+            $('#file-content').parent().addClass('d-none')
+        })
+
+        // Upload the file and display the content
+        $('#file-upload').on('click', function(e) {
+            e.preventDefault()
+            const file = $('#file')[0].files[0]
+            const species = $('#speciesSelect').val()
+            if (file && species) {
+                const formData = new FormData()
+                formData.append('file', file)
+                formData.append('species', species)
+                $.ajax({
+                    type: "POST",
+                    url: "{{ url_for('data.parse_trait_file') }}",
+                    data: formData,
+                    contentType: false,
+                    processData: false,
+                    success: function(response) {
+                        $('.alert').remove()
+                        display_file_content(response)
+                    },
+                    error: function(error){
+                        console.error(error)
+                        $('#file-content').empty()
+                        $('#file-content').parent().addClass('d-none')
+                        $('html').animate({
+                            scrollTop: $('body').offset().top
+                        }, 100);
+                        if (error.responseJSON && error.responseJSON.message) {
+                            $('.alert').remove()
+                            flash_message(error.responseJSON.message)
+                        }         
+                    }
+                })            
+            }            
+        })
+
+        function display_file_content(file) {
```

Comment:
thought -- if briefly squinting at this code, it starts to feel like a....react/vue `component`....:)